### PR TITLE
[MONDRIAN-347] - Incorrect avg calculation in aggregates

### PR DESCRIPTION
**Pull request by akhayrutdinov**
@mkambol, @lucboudreau, review it please

### CODE REVIEW

File: src/main/mondrian/rolap/RolapAggregator.java [hunk 1]
```diff
@@ -270,7 +269,7 @@ public boolean supportsFastAggregates(
      * These aggregators are NOT singletons like the above aggregators; rather,
      * each is different because of the fact count column expression.
      */
-    protected static abstract class BaseAggor extends RolapAggregator {
+    public static abstract class BaseAggor extends RolapAggregator {
```
Comment (lucboudreau) [inline src/main/mondrian/rolap/RolapAggregator.java]: This is not necessary. Move your test in the right package and you won't need to make it public. We don't make things public just to test them like this.

Comment (lucboudreau) [inline src/main/mondrian/rolap/RolapAggregator.java]: Aah I see. You need it in AggStar. Nevermind :)

File: src/main/mondrian/rolap/aggmatcher/AggStar.java [hunk 2]
```diff
@@ -1121,6 +1128,19 @@ public void print(final PrintWriter pw, final String prefix) {
                 pw.print("): ");
                 pw.print(generateRollupString(sqlQuery));
             }
+
+            @Override
+            public String generateExprString(SqlQuery query) {
+                String exprString = super.generateExprString(query);
+                RolapAggregator rollupAggregator = getRollupAggregator();
+                if (rollupAggregator instanceof BaseAggor) {
+                    BaseAggor agg = (BaseAggor) rollupAggregator;
```
Comment (lucboudreau) [inline src/main/mondrian/rolap/aggmatcher/AggStar.java]: This is not a good pattern. Please see if you can make this better. Do we really need the downcast? Can it ever be a different class? Shouldn't the method be part of the RolapAggregator?

Comment (akhayrutdinov) [inline src/main/mondrian/rolap/aggmatcher/AggStar.java]: @lucboudreau, I'm trying to solve a specific case of agg-tables. BaseAggor is declared to be the class for aggregators over agg-tables (https://github.com/pentaho/mondrian/blob/master/src/main/mondrian/rolap/RolapAggregator.java#L265). I cannot figure out how to cram desired changes into ```RolapAggregator``` and even see no sense in it. Do you have any ideas how to implement the solution w/o similar casts?

**Comment by akhayrutdinov** [comment]
@lucboudreau, do you still think the proposal is unacceptable?

**Pull request merged**